Integrations/Elasticsearch_v2/Elasticsearch_v2.py

```diff
@@ -0,0 +1,279 @@
+            if PROXY:
+                es = Elasticsearch(hosts=[SERVER], connection_class=RequestsHttpConnection,
+                                   http_auth=(USERNAME, PASSWORD), verify_certs=INSECURE, proxies=handle_proxy())
+
+            else:
+                es = Elasticsearch(hosts=[SERVER], connection_class=RequestsHttpConnection,
+                                   http_auth=(USERNAME, PASSWORD), verify_certs=INSECURE)
+
+            query = QueryString(query=str(TIME_FIELD) + ":* AND " + FETCH_QUERY)
+            search = Search(using=es, index=FETCH_INDEX).query(query)[0:1]
+            response = search.execute().to_dict()
+            _, total_results = get_total_results(response)
+
+            if total_results == 0:
+                return_error("Fetch incidents test failed - Please check configuration")
+
+            if total_results > 0:
+                hit_date = str(response.get('hits', {}).get('hits')[0].get('_source').get(str(TIME_FIELD)))
+                datetime.strptime(hit_date, TIME_FORMAT).isoformat() + 'Z'
+
+        except ValueError as e:
+            return_error("Inserted time format does not match your index. " + str(e))
+
+    demisto.results('ok')
+
+
+def incident_label_maker(source):
+    labels = []
+    for field in source.keys():
+        labels.append({'type': str(field), 'value': str(source.get(field))})
+
+    return labels
+
+
+def results_to_incidents(response, current_fetch, last_fetch):
+    incidents = []
+    for hit in response.get('hits', {}).get('hits'):
+        if hit.get('_source') is not None and hit.get('_source').get(str(TIME_FIELD)) is not None:
+            hit_date = datetime.strptime(
+                str(hit.get('_source')[str(TIME_FIELD)]), TIME_FORMAT).isoformat() + 'Z'
+            hit_date = datetime.strptime(hit_date, '%Y-%m-%dT%H:%M:%SZ')
+            # update last run
+            if hit_date > last_fetch:
+                last_fetch = hit_date
+
+            # avoid duplication due to weak time query
+            if hit_date > current_fetch:
+                inc = {
+                    'type': 'Elasticsearch',
+                    'sourceBrand': 'Elasticsearch',
+                    'name': 'Elasticsearch: Index: ' + str(FETCH_INDEX) + ", ID: " + str(hit.get('_id')),
+                    'details': json.dumps(hit.get('_source')),
+                    'rawJSON': json.dumps(hit),
+                    'labels': incident_label_maker(hit.get('_source')),
+                    'occurred': hit_date.isoformat() + 'Z'
+                }
+                incidents.append(inc)
+
+    return incidents, last_fetch
+
+
+def fetch_incidents():
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get('time')
+
+    # handle first time fetch
+    if last_fetch is None:
+        last_fetch = datetime.now() - timedelta(days=FETCH_DAYS)
+
+    else:
+        last_fetch = datetime.strptime(last_fetch, '%Y-%m-%dT%H:%M:%SZ')
+
+    current_fetch = last_fetch
+
+    if PROXY:
```

Comment:
You use this bundle of rows 3 times in the code, lets convert it into a function